packages/components/src/internal/components/chart/ChartBuilderModal.tsx

```diff
@@ -0,0 +1,648 @@
+export const MAX_ROWS_PREVIEW = 100000;
+export const MAX_POINT_DISPLAY = 10000;
+const BLUE_HEX_COLOR = '3366FF';
+
+const ICONS = {
+    bar_chart: 'bar_chart',
+    box_plot: 'box_plot',
+    pie_chart: 'pie_chart',
+    scatter_plot: 'xy_scatter',
+    line_plot: 'xy_line',
+};
+
+interface Props extends RequiresModelAndActions {
+    onHide: (successMsg?: string) => void;
+    savedChartModel?: GenericChartModel;
+}
+
+export const ChartBuilderModal: FC<Props> = memo(({ actions, model, onHide, savedChartModel }) => {
+    const CHART_TYPES = LABKEY_VIS?.GenericChartHelper.getRenderTypes();
+    const divId = useMemo(() => generateId('chart-'), []);
+    const ref = useRef<HTMLDivElement>(undefined);
+    const { user } = useServerContext();
+    const canShare = useMemo(
+        () => savedChartModel?.canShare ?? hasPermissions(user, [PermissionTypes.ShareReportPermission]),
+        [savedChartModel, user]
+    );
+    const chartTypes: ChartTypeInfo[] = useMemo(
+        () => CHART_TYPES.filter(type => !type.hidden && !HIDDEN_CHART_TYPES.includes(type.name)),
+        [CHART_TYPES]
+    );
+    const containerFilter = useMemo(() => getContainerFilter(model.containerPath), [model.containerPath]);
+
+    const [loadingData, setLoadingData] = useState<boolean>(false);
+    const [saving, setSaving] = useState<boolean>(false);
+    const [deleting, setDeleting] = useState<boolean>(false);
+    const [showConfirmDelete, setShowConfirmDelete] = useState<boolean>(false);
+    const [error, setError] = useState<string>();
+    const [previewMsg, setPreviewMsg] = useState<string>();
+    const [reportConfig, setReportConfig] = useState<Record<string, any>>();
+    const [selectedType, setSelectedChartType] = useState<ChartTypeInfo>(chartTypes[0]);
+    const [name, setName] = useState<string>('');
+    const [shared, setShared] = useState<boolean>(canShare);
+    const [fieldValues, setFieldValues] = useState<Record<string, SelectInputOption>>({});
+
+    useEffect(
+        () => {
+            if (savedChartModel) {
+                setSelectedChartType(
+                    chartTypes.find(c => savedChartModel?.visualizationConfig.chartConfig.renderType === c.name)
+                );
+                setName(savedChartModel.name);
+                setShared(savedChartModel.shared);
+
+                const measures = savedChartModel.visualizationConfig?.chartConfig?.measures || {};
+                const fieldValues_ = {};
+                Object.keys(measures).map(key => {
+                    let measure = measures[key];
+                    if (measure) {
+                        // Currently only supporting a single measure per axis (i.e. not supporting y-axis left/right)
+                        if (Utils.isArray(measure)) measure = measure[0];
+
+                        fieldValues_[key] = { label: measure.label, value: measure.name, data: measure };
+                    }
+                });
```

Comment:
```suggestion
                const fieldValues_ = Object.keys(measures).reduce((result, key) => {
                    let measure = measures[key];
                    if (measure) {
                        // Currently only supporting a single measure per axis (i.e. not supporting y-axis left/right)
                        if (Utils.isArray(measure)) measure = measure[0];

                        result[key] = { label: measure.label, value: measure.name, data: measure };
                    }
                    return result;
                }, {});
```

We should be using reduce here instead of map, especially since we're ignoring the result of map, and not returning anything within the map function.

Reply:
yup, makes sense. done.